### PR TITLE
WIP: Multipart tagging for objects

### PR DESCRIPTION
**Pull request by suryakumar1024**
Signed-off-by: suryakumar.kumaravelan <suryakumar.kumaravelan@seagate.com>

## Checklist
- Tracker (select at least one)
  - [ ] References tracker ticket
  - [ ] Very recent bug; references commit where it was introduced
  - [ ] New feature (ticket optional)
  - [ ] Doc update (no ticket needed)
  - [ ] Code cleanup (no ticket needed)
- Component impact
  - [ ] Affects [Dashboard](https://tracker.ceph.com/projects/dashboard/issues/new), opened tracker ticket
  - [ ] Affects [Orchestrator](https://tracker.ceph.com/projects/orchestrator/issues/new), opened tracker ticket
  - [ ] No impact that needs to be tracked
- Documentation (select at least one)
  - [ ] Updates relevant documentation
  - [ ] No doc update is appropriate
- Tests (select at least one)
  - [ ] Includes [unit test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/tests-unit-tests/)
  - [ ] Includes [integration test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/testing_integration_tests/)
  - [ ] Includes bug reproducer
  - [ ] No tests

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard cephadm`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`

</details>

### CODE REVIEW

File: src/rgw/rgw_sal_motr.cc [hunk 1]
```diff
@@ -2908,12 +2909,15 @@ int MotrMultipartUpload::init(const DoutPrefixProvider *dpp, optional_yield y,
     // size, etag etc.
     bufferlist bl;
     rgw_bucket_dir_entry ent;
+    MotrObject::Meta meta;
```
Comment (cdeshmukh) [inline src/rgw/rgw_sal_motr.cc:2912]: meta needs to have tags read from request and then encoded within 'ent' i think

File: src/rgw/rgw_sal_motr.cc [hunk 2]
```diff
@@ -2908,12 +2909,15 @@ int MotrMultipartUpload::init(const DoutPrefixProvider *dpp, optional_yield y,
...
     obj->get_key().get_index_key(&ent.key);
     ent.meta.owner = owner.get_id().to_str();
     ent.meta.category = RGWObjCategory::MultiMeta;
     ent.meta.mtime = ceph::real_clock::now();
     ent.meta.user_data.assign(mpbl.c_str(), mpbl.c_str() + mpbl.length());
     ent.encode(bl);
+    encode(attrs, bl);
+    // meta.encode(bl);
```
Comment (cdeshmukh) [inline src/rgw/rgw_sal_motr.cc:2920]: this line should be before bl encloded

File: src/rgw/rgw_sal_motr.cc [hunk 3]
```diff
@@ -3202,10 +3205,16 @@ int MotrMultipartUpload::complete(const DoutPrefixProvider *dpp,
   if (rc < 0) {
     return rc == -ENOENT ? -ERR_NO_SUCH_UPLOAD : rc;
   }
+  rgw::sal::Attrs temp_attrs;
+  // MotrObject::Meta meta;
   rgw_bucket_dir_entry ent;
   bufferlist& blr = bl;
   auto ent_iter = blr.cbegin();
   ent.decode(ent_iter);
+  decode(temp_attrs, ent_iter);
```
Comment (cdeshmukh) [inline src/rgw/rgw_sal_motr.cc:3214]: we may not need anything depending on tests, if we get tags as part of attrs here already

**Review by cdeshmukh — COMMENTED** [review]
@selvakumaar5496 can you too review this, compared to what we did for PutobjTagging API

File: src/rgw/rgw_sal_motr.cc (hunk 1 above)
Comment (SeagateChaDeepak) [inline src/rgw/rgw_sal_motr.cc:2912]: I dont think meta is needed/used. you can remove it.

**Comment by cdeshmukh** [comment]
@suryakumar1024 update PR title to
Tagging with CreateMultipartUpload

And fill up rest of the PR template properly

**Comment by cdeshmukh** [comment]
@suryakumar1024 do we need this still, given you created another PR https://github.com/Seagate/cortx-rgw/pull/235/files
if not, close this one

**Comment by suryakumar1024** [comment]
Closing this PR since raised another PR for the same.
New pR-> https://github.com/Seagate/cortx-rgw/pull/235

**Pull request closed without merging**